server/src/operators/card_operator.rs

```diff
@@ -143,202 +143,129 @@ pub async fn search_card_query(
+                            .map_err(|_| DefaultError {
+                                message: "Failed to parse time range",
+                            })?
+                            .timestamp() as f64,
+                    ),
+                    gte: None,
+                    lte: None,
+                },
             ));
         }
     }
 
     if let Some(serde_json::Value::Object(obj)) = &filters {
         for key in obj.keys() {
-            if let Some(value) = obj.get(key) {
-                match value {
-                    serde_json::Value::Array(arr) => {
-                        if let Some(first_val) = arr.get(0) {
-                            if let Some(string_val) = first_val.as_str() {
-                                query = query.filter(
-                                    sql::<Text>(&format!("card_metadata.metadata->>'{}'", key))
-                                        .ilike(format!("%{}%", string_val)),
-                                );
-                            }
-                        }
-
-                        for item in arr.iter().skip(1) {
-                            if let Some(string_val) = item.as_str() {
-                                query = query.or_filter(
-                                    sql::<Text>(&format!("card_metadata.metadata->>'{}'", key))
-                                        .ilike(format!("%{}%", string_val)),
-                                );
-                            }
-                        }
-                    }
-                    serde_json::Value::String(string_val) => {
-                        query = query.filter(
-                            sql::<Text>(&format!("card_metadata.metadata->>'{}'", key))
-                                .ilike(format!("%{}%", string_val)),
-                        );
-                    }
-                    _ => (),
+            let value = obj.get(key).expect("Value should exist");
+            match value {
+                serde_json::Value::Array(arr) => {
+                    filter.must.push(Condition::matches(
+                        &format!("metadata.{}", key),
+                        arr.iter()
+                            .map(|item| item.to_string())
+                            .collect::<Vec<String>>(),
+                    ));
+                }
+                _ => {
+                    filter.must.push(Condition::matches(
+                        &format!("metadata.{}", key),
+                        value.to_string().replace('\"', ""),
+                    ));
                 }
             }
         }
     }
 
     if let Some(quote_words) = parsed_query.quote_words {
         for word in quote_words.iter() {
-            query = query.filter(card_metadata_columns::content.ilike(format!("%{}%", word)));
+            filter
+                .must
+                .push(Condition::matches("card_html", word.clone() + " "));
```

Comment:
```suggestion
                .push(Condition::matches("card_html", word.clone()));
```

I don't think you need the quote here. It's fine if it's a sub-word imo.

Reply:
>Yeah no there has to be a white space or else it doesn’t do a substring search

Per Matrix